File: frontend/EduMatch/src/types/payment.ts
export type PaymentStatus = "pending" | "held" | "released" | "refunded" | "cancelled" | "disputed";
export type PaymentMethod = "card_simulation" | "wallet_simulation" | "cash_simulation";

export interface Payment {
  id: number;
  session_id: number;
  request_id: number;
  student_id: number;
  instructor_id: number;
  amount: string;
  platform_fee: string;
  total_amount: string;
  status: PaymentStatus;
  payment_method: PaymentMethod;
  paid_at: string | null;
  released_at: string | null;
  refunded_at: string | null;
  created_at: string;
  updated_at: string;
  request_title: string | null;
  student_name: string | null;
  instructor_name: string | null;
  session_status?: string | null;
  request_status?: string | null;
}

File: backend/app/models/payment.py
from datetime import datetime
from decimal import Decimal

from sqlalchemy import DateTime, ForeignKey, Numeric, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Payment(Base):
    __tablename__ = "payments"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    session_id: Mapped[int] = mapped_column(ForeignKey("sessions.id", ondelete="CASCADE"), index=True, nullable=False)
    request_id: Mapped[int] = mapped_column(ForeignKey("requests.id", ondelete="CASCADE"), index=True, nullable=False)
    student_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False)
    instructor_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False)
    amount: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    platform_fee: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    total_amount: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    status: Mapped[str] = mapped_column(String(32), nullable=False, default="pending", server_default="pending")
    payment_method: Mapped[str] = mapped_column(String(32), nullable=False, default="card_simulation", server_default="card_simulation")
    paid_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    released_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    refunded_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    session = relationship("Session", back_populates="payments")
    request = relationship("LearningRequest", back_populates="payments")
    student = relationship("User", foreign_keys=[student_id], back_populates="student_payments")
    instructor = relationship("User", foreign_keys=[instructor_id], back_populates="instructor_payments")
    wallet_transactions = relationship("WalletTransaction", back_populates="payment", cascade="all, delete-orphan")
